client-ios/actions/index.js

```diff
@@ -41,14 +34,35 @@ export const updateUserInputCategory = (category) => {
 }
 
 // updates the global state allItems from the userInput
+//REMOVE?
 export const addNewListItem = () => {
   return {
     type: types.ADD_NEW_LIST_ITEM
   }
 }
 
+export const mainButtonPressed = (buttonCategory) => {
+  return (dispatch, getState) => {
+
+    let userInput = getState().lists.userInput;
+    if (userInput) {
+      let newItem = {
+        title: userInput,
+        category: buttonCategory
+      }
+      console.log('category', buttonCategory);
+
+      dispatch(addItemLocally(newItem));
+      dispatch(addItemToDatabase(newItem));
+    }
+    else {
+      dispatch(updateFilter(buttonCategory));
+      Actions.singleListScreen();
+    }
+  }
+}
 // REFACTORED version
-export const addItem = (item) => {
+export const addItemToDatabase = (item) => {
   return (dispatch, getState) => {
     dispatch(addNewListItemDatabaseRequest());
     let user = getState().lists.user;
```

Comment:
you'd better change getState().lists.user to getState().auth.user because Brent changed global state design a little bit!